perfact/zodbsync/zodbsync.py

```diff
@@ -162,6 +164,9 @@ def mod_write(data, parent=None, obj_id=None, override=False, root=None,
     for handler in mod_implemented_handlers(obj, meta_type):
         handler.write(obj, d)
 
+    # Also write zodbsync layer information
+    obj.zodbsync_layer = layer
```

Comment:
I'm pretty sure this will have no effect on most `record` runs since they don't commit their transaction. In particular, the `watch` command explicitly aborts its transaction (https://github.com/perfact/zodbsync/blob/main/perfact/zodbsync/commands/watch.py#L357) since it is used to only make sure a consistent state is viewed and it does not expect that anything is changed in the Data.FS. And a regular `record` run uses an implicit transaction that is neither commited nor aborted, which I am pretty sure means an implicit abort.